projects/ngx-formentry/src/form-entry/helpers/js-expression-helper.ts

```diff
@@ -169,6 +169,15 @@ export class JsExpressionHelper {
 
 
   calcSouthEastAsiaNonLabCVDRisk(sex: 'M' | 'F', smoker: boolean, age: number, sbp: number, bmi: number) {
+    const hasMissingValue = (sex === null || typeof sex === "undefined") ||
+      (smoker === null || typeof smoker === "undefined") ||
+      (isNaN(age) || typeof age === "undefined") ||
+      (isNaN(sbp) || typeof sbp === "undefined") ||
+      (isNaN(bmi) || typeof bmi === "undefined");
```

Comment:
I'd be careful with `isNaN()`... we still probably need an explicit `null` check. Maybe it would be better to do positive assertions, e.g.,

`const hasValidValues = (typeof sex === "string" && typeof smoker === "boolean" && typeof age === "number" /* etc. */);`

What do you think?

Reply:
Sounds good let me try that.

Reply:
@ibacher I have used the `positive` conditioning and seems to be more reliable.